pkg/tuple/tuple_test.go

```diff
@@ -297,6 +299,39 @@ var testCases = []struct {
 		tupleFormat:    nil,
 		relFormat:      nil,
 	},
+	{
+		input:          "testns:" + superLongID + "#testrel@user:testusr",
+		expectedOutput: "testns:" + superLongID + "#testrel@user:testusr",
+		tupleFormat: makeTuple(
+			ObjectAndRelation("testns", superLongID, "testrel"),
+			ObjectAndRelation("user", "testusr", "..."),
+		),
+		relFormat: rel("testns", superLongID, "testrel", "user", "testusr", ""),
+	},
+	{
+		input:          "testns:foo#testrel@user:" + superLongID,
+		expectedOutput: "testns:foo#testrel@user:" + superLongID,
+		tupleFormat: makeTuple(
+			ObjectAndRelation("testns", "foo", "testrel"),
+			ObjectAndRelation("user", superLongID, "..."),
+		),
+		relFormat: rel("testns", "foo", "testrel", "user", superLongID, ""),
+	},
+	{
+		input:          "testns:foo#testrel@user:" + superLongID + "more",
+		expectedOutput: "",
+		tupleFormat:    nil,
+		relFormat:      nil,
+	},
+	{
+		input:          "testns:-base64YWZzZGZh-ZHNmZHPwn5iK8J+YivC/fmIrwn5iK==#testrel@user:-base65YWZzZGZh-ZHNmZHPwn5iK8J+YivC/fmIrwn5iK==",
+		expectedOutput: "testns:-base64YWZzZGZh-ZHNmZHPwn5iK8J+YivC/fmIrwn5iK==#testrel@user:-base65YWZzZGZh-ZHNmZHPwn5iK8J+YivC/fmIrwn5iK==",
+		tupleFormat: makeTuple(
+			ObjectAndRelation("testns", "-base64YWZzZGZh-ZHNmZHPwn5iK8J+YivC/fmIrwn5iK==", "testrel"),
+			ObjectAndRelation("user", "-base65YWZzZGZh-ZHNmZHPwn5iK8J+YivC/fmIrwn5iK==", "..."),
+		),
+		relFormat: rel("testns", "-base64YWZzZGZh-ZHNmZHPwn5iK8J+YivC/fmIrwn5iK==", "testrel", "user", "-base65YWZzZGZh-ZHNmZHPwn5iK8J+YivC/fmIrwn5iK==", ""),
+	},
```

Comment:
it seems like we are testing only the valid cases, but how can we be sure the validation is working if we are not testing an invalid tuple?

Reply:
We test invalid cases above